flight/uavobjectlib/Makefile

```diff
@@ -66,8 +66,15 @@ LDFLAGS += -Wl,--fatal-warnings
 SRC += $(filter-out $(OPUAVSYNTHDIR)/uavobjectsinit.c,$(wildcard $(OPUAVSYNTHDIR)/*.c))
 NO_AUTO_UAVO = YES
 
+# Backup the environmental variable before entering firmware-common.mk
```

Comment:
If you set an environmental variable on the command line, then it is passed to all Makefiles and cannot be overwritten without the `override` command. That means that a command such as `make RTOS=FREERTOS quanton` will cause a compile failure without this save/restore.